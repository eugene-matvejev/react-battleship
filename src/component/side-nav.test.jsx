import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SideNav from './side-nav';

configure({ adapter: new Adapter() });

describe('<SideNav/>', () => {
    const props = {
        title: '{{title}}',
        routes: [
            { to: '/1', label: 'label1', disabled: true },
            { to: '/2', label: 'label2', disabled: false },
        ],
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<SideNav {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['className', '{{className}}'],
                ['data-cy', '{{data-cy}}'], /** should be passed 'as is' */
                ['isCollapsed', true],
                ['isCollapsed', false],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<SideNav {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });

    describe('internal callbacks', () => {
        describe('::onCollapse', () => {
            it('should toggle state field [::isCollapsed] from a click on [data-cy="sidenav-collapse-button"]', () => {
                const spy = spyOn(SideNav.prototype, 'setState');

                const c = shallow(<SideNav {...props}/>);
                c.find('[data-cy="sidenav-collapse-button"]').simulate('click');

                expect(spy).toBeCalledWith({ isCollapsed: !c.state('isCollapsed') });
            });
        });
    });
});
