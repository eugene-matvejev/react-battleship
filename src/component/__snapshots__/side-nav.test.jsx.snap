// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav/> render with default/required props 1`] = `
<aside
  className="sidenav "
>
  <button
    className="sidenav_collapse-button"
    data-cy="sidenav-collapse-button"
    onClick={[Function]}
  />
  <h2
    className="sidenav_title"
    data-cy="sidenav-label"
  >
    {{title}}
  </h2>
  <NavLink
    activeClassName="sidenav_link--active"
    className="sidenav_link--disabled"
    data-cy="sidenav-link-0"
    disabled={true}
    key="0"
    to="/1"
  >
    label1
  </NavLink>
  <NavLink
    activeClassName="sidenav_link--active"
    className="sidenav_link"
    data-cy="sidenav-link-1"
    disabled={false}
    key="1"
    to="/2"
  >
    label2
  </NavLink>
</aside>
`;

exports[`<SideNav/> render with optional props [::className] as "{{className}}" 1`] = `
<aside
  className="sidenav {{className}}"
>
  <button
    className="sidenav_collapse-button"
    data-cy="sidenav-collapse-button"
    onClick={[Function]}
  />
  <h2
    className="sidenav_title"
    data-cy="sidenav-label"
  >
    {{title}}
  </h2>
  <NavLink
    activeClassName="sidenav_link--active"
    className="sidenav_link--disabled"
    data-cy="sidenav-link-0"
    disabled={true}
    key="0"
    to="/1"
  >
    label1
  </NavLink>
  <NavLink
    activeClassName="sidenav_link--active"
    className="sidenav_link"
    data-cy="sidenav-link-1"
    disabled={false}
    key="1"
    to="/2"
  >
    label2
  </NavLink>
</aside>
`;

exports[`<SideNav/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<aside
  className="sidenav "
>
  <button
    className="sidenav_collapse-button"
    data-cy="{{data-cy}}sidenav-collapse-button"
    onClick={[Function]}
  />
  <h2
    className="sidenav_title"
    data-cy="{{data-cy}}sidenav-label"
  >
    {{title}}
  </h2>
  <NavLink
    activeClassName="sidenav_link--active"
    className="sidenav_link--disabled"
    data-cy="{{data-cy}}sidenav-link-0"
    disabled={true}
    key="0"
    to="/1"
  >
    label1
  </NavLink>
  <NavLink
    activeClassName="sidenav_link--active"
    className="sidenav_link"
    data-cy="{{data-cy}}sidenav-link-1"
    disabled={false}
    key="1"
    to="/2"
  >
    label2
  </NavLink>
</aside>
`;

exports[`<SideNav/> render with optional props [::isCollapsed] as "false" 1`] = `
<aside
  className="sidenav "
>
  <button
    className="sidenav_collapse-button"
    data-cy="sidenav-collapse-button"
    onClick={[Function]}
  />
  <h2
    className="sidenav_title"
    data-cy="sidenav-label"
  >
    {{title}}
  </h2>
  <NavLink
    activeClassName="sidenav_link--active"
    className="sidenav_link--disabled"
    data-cy="sidenav-link-0"
    disabled={true}
    key="0"
    to="/1"
  >
    label1
  </NavLink>
  <NavLink
    activeClassName="sidenav_link--active"
    className="sidenav_link"
    data-cy="sidenav-link-1"
    disabled={false}
    key="1"
    to="/2"
  >
    label2
  </NavLink>
</aside>
`;

exports[`<SideNav/> render with optional props [::isCollapsed] as "true" 1`] = `
<aside
  className="sidenav--collapsed "
>
  <button
    className="sidenav_collapse-button"
    data-cy="sidenav-collapse-button"
    onClick={[Function]}
  />
</aside>
`;
